instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EVRouteRefreshTest.kt

```diff
@@ -349,6 +351,53 @@ class EVRouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.
         )
     }
 
+    @Test
+    fun ev_route_refresh_updates_waypoints_per_route() = sdkTest {
+        replaceOriginalResponseHandler(R.raw.ev_route_response_for_refresh_with_waypoints_per_route)
+        addRefreshRequestHandler(
+            R.raw.ev_route_refresh_response,
+            acceptedGeometryIndex = 0,
+            testUuid = "ev_route_response_for_refresh_with_waypoints_per_route"
+        )
+        val requestedRoutes = requestRoutes(
+            twoCoordinates,
+            electric = true,
+            waypointsPerRoute = true
+        )
+        val evData = mapOf(
+            KEY_ENERGY_CONSUMPTION_CURVE to "0,300;20,160;80,140;120,180",
+            KEY_EV_INITIAL_CHARGE to "17000",
+            KEY_EV_PRECONDITIONING_TIME to "10",
+            KEY_AUXILIARY_CONSUMPTION to "300"
+        )
+        mapboxNavigation.onEVDataUpdated(evData)
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(requestedRoutes)
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+        val updatedRoutes = waitUntilRefresh().navigationRoutes
```

Comment:
we set route created with charge level `18000` to navigation which already has charge level `17000`. Is that expected?

Reply:
Sure. The routes were requested when the charge was 18000. Then the charge dropped to 17000 and we told MapboxNavigation about it. Then we set the requested routes. I don't see anything invalid in this case. The latest data (17000) should be used in the refresh request.